contrib/src/databases.rs

```diff
@@ -0,0 +1,42 @@
+pub extern crate r2d2;
+
+use std::marker::{Send, Sized};
+use rocket::config;
+
+pub use self::r2d2::*;
+
+#[cfg(feature = "diesel_sqlite")]
+pub extern crate diesel;
+
+#[cfg(feature = "pg_pool")]
+pub extern crate postgres;
+#[cfg(feature = "pg_pool")]
+pub extern crate r2d2_postgres;
+
+pub trait Poolable
+    where Self: 'static + Sized + Send {
+    type Manager: ManageConnection<Connection=Self>;
+
+    fn manager(info: config::Table) -> Self::Manager;
+}
```

Comment:
@SergioBenitez Just a heads up: we're going to run into orphaned impl rules with this. I ran into this in assembling this PR. A consuming library of this trait, can't implement it on a foreign type.

More specifically, I can't `impl Poolable for diesel::SqliteConnection` outside of the `databases` module in the `contrib` crate. So this doesn't really solve the issue of user extensibility, but it does make it easy to instantiate arbitrary pools.

Maybe I'm missing something, but it doesn't seem immediately obvious to me.

Reply:
Ah. I misunderstood our desire for user extensibility. I thought it was intended for an arbitrary user, not a database library author. Makes sense and will totally work given our strategy!